lib/Sabberworm/CSS/Parsing/ParserState.php

```diff
@@ -48,13 +50,26 @@ public function getSettings() {
 		return $this->oParserSettings;
 	}
 
+	public function setAnchor() {
+		$this->iAnchor = $this->iCurrentPosition;
+	}
+
+	public function backtrackToAnchor() {
+		if ($this->iAnchor !== null) {
+			$this->iCurrentPosition = $this->iAnchor;
+		}
+	}
+
```

Comment:
I kinda like the idea with the anchor but I’d prefer it to be implemented as a separate class:

```php
class Anchor {
	private $iPosition;
	private $oParserState; 
[…]
	function backtrack() {
		$this->oParserState->iCurrentPosition = $this->iPosition; // Or using setter
	}
}

class ParserState {
[…]
	function anchor() {
		return new Anchor($this->iCurrentPosition, $this);
	}
[…]
}